crates/uv-settings/src/settings.rs

```diff
@@ -136,24 +136,237 @@ pub struct ResolverInstallerOptions {
         "#
     )]
     pub index_url: Option<IndexUrl>,
+    /// Extra URLs of package indexes to use, in addition to `--index-url`.
+    ///
+    /// Accepts either a repository compliant with PEP 503 (the simple repository API), or a local
+    /// directory laid out in the same format.
+    ///
+    /// All indexes provided via this flag take priority over the index specified by
+    /// [`index_url`](#index-url). When multiple indexes are provided, earlier values take priority.
+    #[option(
+        default = "[]",
+        value_type = "list[str]",
+        example = r#"
+            extra-index-url = ["https://download.pytorch.org/whl/cpu"]
+        "#
+    )]
     pub extra_index_url: Option<Vec<IndexUrl>>,
+    /// Ignore all registry indexes (e.g., PyPI), instead relying on direct URL dependencies and
+    /// those provided via `--find-links`.
+    #[option(
+        default = "false",
+        value_type = "bool",
+        example = r#"
+            no-index = true
+        "#
+    )]
     pub no_index: Option<bool>,
+    /// Locations to search for candidate distributions, in addition to those found in the registry
+    /// indexes.
+    ///
+    /// If a path, the target must be a directory that contains packages as wheel files (`.whl`) or
+    /// source distributions (`.tar.gz` or `.zip`) at the top level.
+    ///
+    /// If a URL, the page must contain a flat list of links to package files adhering to the
+    /// formats described above.
+    #[option(
+        default = "[]",
+        value_type = "list[str]",
+        example = r#"
+            find-links = ["https://download.pytorch.org/whl/torch_stable.html"]
+        "#
+    )]
     pub find_links: Option<Vec<FlatIndexLocation>>,
+    /// The strategy to use when resolving against multiple index URLs.
+    ///
+    /// By default, `uv` will stop at the first index on which a given package is available, and
+    /// limit resolutions to those present on that first index (`first-match`). This prevents
+    /// "dependency confusion" attacks, whereby an attack can upload a malicious package under the
+    /// same name to a secondary.
+    ///
+    /// Possible values:
+    ///
+    /// - `"first-index"`:        Only use results from the first index that returns a match for a given package name.
+    /// - `"unsafe-first-match"`: Search for every package name across all indexes, exhausting the versions from the first index before moving on to the next.
+    /// - `"unsafe-best-match"`:  Search for every package name across all indexes, preferring the "best" version found. If a package version is in multiple indexes, only look at the entry for the first index.
```

Comment:
Stopped adding these because I'm going to explore auto-generating them in the macro (separately).